metrics/internal/collectors/ceph-object-store_test.go

```diff
@@ -0,0 +1,294 @@
+package collectors
+
+import (
+	"testing"
+
+	"github.com/openshift/ocs-operator/metrics/internal/options"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	cephv1listers "github.com/rook/rook/pkg/client/listers/ceph.rook.io/v1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	mockOpts = &options.Options{
+		Apiserver:         "https://localhost:8443",
+		KubeconfigPath:    "",
+		Host:              "0.0.0.0",
+		Port:              8080,
+		ExporterHost:      "0.0.0.0",
+		ExporterPort:      8081,
+		AllowedNamespaces: []string{"openshift-storage"},
+		Help:              false,
+	}
+	mockCephObjectStore1 = cephv1.CephObjectStore{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "ceph.rook.io/v1",
+			Kind:       "CephObjectStore",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mockCephObjectStore-1",
+			Namespace: "openshift-storage",
+		},
+		Spec:   cephv1.ObjectStoreSpec{},
+		Status: &cephv1.ObjectStoreStatus{},
+	}
+	mockCephObjectStore2 = cephv1.CephObjectStore{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mockCephObjectStore-2",
+			Namespace: "openshift-storage",
+		},
+		Spec:   cephv1.ObjectStoreSpec{},
+		Status: &cephv1.ObjectStoreStatus{},
+	}
+	mockCephObjectStore3 = cephv1.CephObjectStore{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mockCephObjectStore-3",
+			Namespace: "default",
+		},
+		Spec:   cephv1.ObjectStoreSpec{},
+		Status: &cephv1.ObjectStoreStatus{},
+	}
+)
+
+func setKubeConfig(t *testing.T) {
+	kubeconfig, err := clientcmd.BuildConfigFromFlags(mockOpts.Apiserver, mockOpts.KubeconfigPath)
+	assert.Nil(t, err, "error: %v", err)
+
+	mockOpts.Kubeconfig = kubeconfig
+}
+
+func getMockCephObjectStoreCollector(t *testing.T, mockOpts *options.Options) (mockCephObjectStoreCollector *CephObjectStoreCollector) {
+	setKubeConfig(t)
+	mockCephObjectStoreCollector = NewCephObjectStoreCollector(mockOpts)
+	assert.NotNil(t, mockCephObjectStoreCollector)
+	return
+}
+
+func setInformerStore(t *testing.T, objs []*cephv1.CephObjectStore, mockCephObjectStoreCollector *CephObjectStoreCollector) {
+	if objs != nil {
+		for _, obj := range objs {
+			err := mockCephObjectStoreCollector.Informer.GetStore().Add(obj)
+			assert.Nil(t, err)
+		}
+	}
+}
+
+func resetInformerStore(t *testing.T, objs []*cephv1.CephObjectStore, mockCephObjectStoreCollector *CephObjectStoreCollector) {
+	if objs != nil {
+		for _, obj := range objs {
+			err := mockCephObjectStoreCollector.Informer.GetStore().Delete(obj)
+			assert.Nil(t, err)
+		}
+	}
+}
+
+func TestNewCephObjectStoreCollector(t *testing.T) {
+	type args struct {
+		opts *options.Options
+	}
+	tests := []struct {
+		name string
+		args args
```

Comment:
Yes! I love seeing case tables from the get-go. :) But, instead of creating a separate `args` struct, just define the arguments as part of the main struct.

Reply:
This was auto-generated as part of standard go unit test generation. And, I retained it as it seemed cleaner to keep `args` isolated from rest of the fields.

Reply:
All right, I'll take it.